src/meta/src/stream/source_manager.rs

```diff
@@ -71,7 +71,8 @@ pub struct SourceManager<S: MetaStore> {
     catalog_manager: CatalogManagerRef<S>,
     barrier_scheduler: BarrierScheduler<S>,
     compaction_group_manager: CompactionGroupManagerRef<S>,
-    core: Arc<Mutex<SourceManagerCore<S>>>,
+    core: Mutex<SourceManagerCore<S>>,
+    paused: Mutex<bool>,
```

Comment:
Does `AtomicBool` with `AcqRel` or `SeqCst` work here?

Reply:
There may be a long wait here, for example, if the source manager is pushing split change barrier, the process may be long and using atomic may cause the cpu to spin for a long time, or is there a better way to use it?

Reply:
How about using a `Mutex<()>`? To pause the ticking during scaling, we can simply hold a guard to this mutex, which will be automatically dropped after success or failure. Besides, the source manager will hold the guard during the whole process of **every** tick.

Reply:
Yes, this will be much easier, I'll change it